### PR TITLE
Provider/Azure: Add cache for VMSS.

### LINKED ISSUE #83073: [Azure] EnsureHostInPool TooManyRequests
**What happened**:
Recently we upgraded our cluster from 1.14.4 to 1.14.7. Then we started to observe tons of following errors:
```
E0924 07:21:12.021680       1 service_controller.go:219] error processing service ingress-nginx/ingress-nginx126 (will retry): failed to ensure load balancer for service ingress-nginx/ingress-nginx126: [EnsureHostInPool(ingress-nginx/ingress-nginx126): backendPoolID(/subscriptions/***/resourceGroups/***/providers/Microsoft.Network/loadBalancers/srprodacseastusb/backendAddressPools/***) - failed to ensure host in pool: "compute.VirtualMachineScaleSetVMsClient#Get: Failure responding to request: StatusCode=429 -- Original Error: autorest/azure: Service returned an error. Status=429 Code=\"OperationNotAllowed\" Message=\"The server rejected the request because too many requests have been received for this subscription.\" Details=[{\"code\":\"TooManyRequests\",\"message\":\"{\\\"operationGroup\\\":\\\"GetVMScaleSetVM3Min\\\",\\\"startTime\\\":\\\"2019-09-24T07:21:11.6439379+00:00\\\",\\\"endTime\\\":\\\"2019-09-24T07:23:11.6439379+00:00\\\",\\\"allowedRequestCount\\\":500,\\\"measuredRequestCount\\\":534}\",\"target\":\"GetVMScaleSetVM3Min\"}] InnerError={\"internalErrorCode\":\"TooManyRequestsReceived\"}", .....
```

I guess the cause may be related to this line which was introduced recently:
https://github.com/kubernetes/kubernetes/blob/6f0f05150028dcaad4c3b5001d84b230502c16e6/pkg/cloudprovider/providers/azure/azure_vmss.go#L805

As we have hundreds of services with LoadBalancer type, and we also enable cluster-autoscaler, the nodes list change very often. For each service, there will be one call to GetScaleSet. The quota will be exhausted quickly. When `EnsureHostInPool` fails, new nodes created by cluster-autoscaler will lose connectivity to load balancer.

**What you expected to happen**:
Kubernetes should not hit VMSS throttling threshold.

**How to reproduce it (as minimally and precisely as possible)**:
1. Provision a cluster with aks-engine. Use VMSS.
2. Create 200 services with LoadBalancer type.
3. Scale up VMSS to 50 nodes
4. Scale down VMSS to 60 nodes
5. Repeat scaling up and down VMSS

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.14.7
- Cloud provider or hardware configuration: Azure, VMSS
- OS (e.g: `cat /etc/os-release`): Ubuntu
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind bug

**What this PR does / why we need it**:

Ref: https://github.com/kubernetes/kubernetes/issues/83073

In some specific scenario, VMSS.Get will throttle, causing `EnsureHostInPool ` to fail. A VMSS cache is added to mitigate this issue.

**Which issue(s) this PR fixes**:

Fixes: https://github.com/kubernetes/kubernetes/issues/83073

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
VMSS cache is added so that less chances of VMSS GET throttling
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/area provider/azure

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go [hunk 1]
```diff
@@ -50,6 +53,57 @@ type vmssVirtualMachinesEntry struct {
...
+			}
+		}
+
+		return localCache, nil
+	}
+
+	return newTimedcache(vmssTTL, getter)
+}
+
+func (ss *scaleSet) deleteCacheForVMSS(vmssName string) error {
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: please delete this function as it is not referenced any where.

**Review by feiskyer — APPROVED** [review]
Looks good. Let's run the e2e tests.

**Comment by andyzhangx** [comment]
it's conflict now, pls fix it

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go [hunk 2]
```diff
@@ -48,6 +52,49 @@ type vmssVirtualMachinesEntry struct {
...
+			}
+		}
+
+		return localCache, nil
+	}
+
+	return newTimedcache(vmssTTL, getter)
+}
+
+func (ss *scaleSet) makeVmssVMName(scaleSetName, instanceID string) string {
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: why adding this? rebase issues?

Comment (nilo19) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: yeah created during rebase. Will delete it.

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go (hunk 2 above)
Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: 👍

**Comment by andyzhangx** [comment]
@nilo19 also need to cherry-pick to 1.17, thanks.

**Pull request merged**